controllers/glanceapi_controller.go

```diff
@@ -1320,6 +1330,36 @@ func (r *GlanceAPIReconciler) deleteJob(
 	return ctrlResult, err
 }
 
+// getEndpointID - returns the endpointID associated to a keystone Endpoint
+func (r *GlanceAPIReconciler) getEndpointID(
+	ctx context.Context,
+	instance *glancev1.GlanceAPI,
+) (string, error) {
+	ep := &keystonev1.KeystoneEndpoint{}
+	epID := ""
+	epType := endpoint.EndpointInternal
+	// in case of split, the external API will point to its own keystone Endpoint
+	// instead of looking for a different API that might not exist or have issues
+	// we can't control here
+	if instance.Spec.APIType == glancev1.APIExternal {
+		epType = endpoint.EndpointPublic
+	}
+	err := r.Get(ctx, types.NamespacedName{Namespace: instance.Namespace, Name: instance.Name}, ep)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			// Just log the keystoneEndpoint CR does not exist so we have evidence
+			// in the operator output: it's not necessarily an error that should
+			// trigger a reconciliation loop
+			r.Log.Info(fmt.Sprintf("EndpointID not found for glanceAPI %s", instance.Name))
+		}
+		return epID, nil
```

Comment:
here epID will be blank, right?
And why we are returning err as nil, just to trigger reconcile loop?

Reply:
Yes, L1339 explicitly initialize it to an empty string, also we do not return err because it might happen that a different glanceAPI (not registered in keystone) is reconciling: in this case it won't find a `keystoneEndpoint` CR and we shouldn't block the deployment because `epID` is not available.